src/data/nat/periodic.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2021 Bolton Bailey. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Bolton Bailey
+-/
+import algebra.periodic
+import data.nat.interval
+import data.nat.count
+
+/-!
+# Periodic Functions on ℕ
+
+This file identifies a few functions on `ℕ` which are periodic, and also proves a lemma about
+periodic predicates which helps determine their cardinality when filtering intervals over them.
+-/
+
+open finset nat function
+
+lemma periodic_gcd (a : ℕ) : periodic (gcd a) a :=
+by simp only [forall_const, gcd_add_self_right, eq_self_iff_true, periodic]
+
+lemma periodic_coprime (a : ℕ) : periodic (coprime a) a :=
+by simp only [coprime_add_self_right, forall_const, iff_self, eq_iff_iff, periodic]
+
+lemma periodic_mod (a : ℕ) : periodic (λ n, n % a) a :=
+by simp only [forall_const, eq_self_iff_true, add_mod_right, periodic]
+
+/-- An interval of length `a` filtered over a periodic predicate of period `a` has the
+same cardinality as `range a` filtered over that predicate. -/
```

Comment:
This docstring isn't saying what the type says (even though it is of course correct)